### PR TITLE
[Vpc Mgr] Register and Unregister VPC with GWM

### PR DESCRIPTION
**Pull request by songxiaoyan**
- Register VPC with GM
- UnRegister VPC with GM
- Update the deleteGatewayInfoForZeta method of GatewayService in GM

### CODE REVIEW

**Review by cj-chung — CHANGES_REQUESTED** [review]
@songxiaoyan Most of code look good. Can you add one more function for unregister VPC from Gateway Manager when user delete a VPC. I have already added a workflow for unregister a VPC in the gateway design doc, please refer to https://github.com/cj-chung/alcor/blob/gatewaydesign/docs/modules/ROOT/pages/mgmt_services/gateway_manager.adoc

**Comment by songxiaoyan** [comment]
> @songxiaoyan Most of code look good. Can you add one more function for unregister VPC from Gateway Manager when user delete a VPC. I have already added a workflow for unregister a VPC in the gateway design doc, please refer to https://github.com/cj-chung/alcor/blob/gatewaydesign/docs/modules/ROOT/pages/mgmt_services/gateway_manager.adoc

Ok, I will do it

File: services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java [hunk 1]
```diff
@@ -102,4 +108,19 @@ public boolean checkSubnetsAreEmpty(VpcEntity vpcEntity) throws SubnetsNotEmptyE
...
+        VpcInfo vpcInfo = new VpcInfo(vpcEntity.getId(), vpcEntity.getSegmentationId(), vpcEntity.getProjectId());
+        HttpEntity<VpcInfoJson> vpcHttpEntity = new HttpEntity<>(new VpcInfoJson(vpcInfo));
+        return restTemplate.postForObject(url, vpcHttpEntity, ResponseId.class);
+    }
+
+    @Override
+    public ResponseId unRegisterVpc(VpcEntity vpcEntity) {
+        String url = gatewayUrl + "project/" + vpcEntity.getProjectId() + "/gatewayinfo/" + vpcEntity.getId();
+        restTemplate.delete(url);
+        return new ResponseId(vpcEntity.getId());
```
Comment (cj-chung) [inline services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java:124]: @songxiaoyan did you handle workflow for unregister vpc in gateway manager which is sending notification to zeta and dpm?

Comment (cj-chung) [reply services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java:124]: and remove attachment and gateway entry in gateway manager.

Comment (songxiaoyan) [reply services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java:124]: I already did it in the GatewayManager, you can review code in this PR.

Comment (songxiaoyan) [reply services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java:124]: I already did it in the GatewayManager, you can review code in this PR.

**Review by cj-chung — APPROVED** [review]
LGTM

File: services/vpc_manager/src/main/resources/application.properties [hunk 2]
```diff
@@ -2,6 +2,7 @@ server.port=9001
 #####Microservice url configuration######
 microservices.subnet.service.url=http://localhost:9002
 microservices.route.service.url=http://localhost:9003/
+microservices.gateway.service.url=http://localhost:9015/
```
Comment (xieus) [inline services/vpc_manager/src/main/resources/application.properties:5]: @songxiaoyan We can also add this configuration to the yaml file

https://github.com/futurewei-cloud/alcor/blob/master/kubernetes/services/vpc_manager.yaml

**Review by xieus — APPROVED** [review]
LGTM

**Pull request merged**